features/call/src/main/kotlin/io/element/android/features/call/CallIntentDataParser.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.call
+
+import android.net.Uri
+import java.net.URLDecoder
+
+object CallIntentDataParser {
+
+    private val validHttpSchemes = sequenceOf("http", "https")
+
+    fun parse(data: String?): String? {
+        val parsedUrl = data?.let { Uri.parse(data) } ?: return null
+        val scheme = parsedUrl.scheme
+        return when {
+            scheme in validHttpSchemes && parsedUrl.host == "call.element.io" -> data
+            scheme == "element" && parsedUrl.host == "call" -> {
+                parsedUrl.getQueryParameter("url")
+                    ?.let { URLDecoder.decode(it, "utf-8") }
+                    ?.takeIf { Uri.parse(it)?.scheme in validHttpSchemes }
```

Comment:
You do not check the host here as per line 30? This is maybe OK, but a comment would be welcome if this is the case, and a test case could be added to cover the case.

Reply:
This is for opening arbitrary urls through `element://call?url=https://some-custom-ec-instance.com/actual-call` so the best we could do is check it's not null or blank, I think.

Reply:
That's what I was suspected. Thanks!